buildroot/rebuild.sh

```diff
@@ -1,44 +1,62 @@
 #!/usr/bin/env bash
-# Takes the architecture as a parameter
-# Currently supported values: arm64
+# Takes a list of architectures to build images for as the parameter
 
 # Make sure we don't have any unset variables
 set -u
 
-# Clean up
-rm -rf build
-mkdir -p build
+# Move into the folder that contains this script
+cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" || exit 1
+
+# Generate list of configs to build
+CONFIGS=()
+while (( ${#} )); do
+    case ${1} in
+        all) for CONFIG in *.config; do CONFIGS+=( "../${CONFIG}" ); done ;;
+        arm64|arm|ppc32|ppc64le|x86_64) CONFIGS+=( "../${1}.config" ) ;;
+        *) echo "Unknown parameter '${1}', exiting!"; exit 1 ;;
+    esac
+    shift
+done
 
 # Download latest buildroot release
-curl https://buildroot.org/downloads/buildroot-2019.02.3.tar.gz | tar -xzf - -C build --strip-components=1
-cd build || exit 1
-
-# Use the config in the parent folder
-CONFIG=../${1}.config
-if [[ ! -f ${CONFIG} ]]; then
-    echo "${CONFIG} does not exist! Is your parameter correct?"
-    exit 1
-fi
-BR2_DEFCONFIG=${CONFIG} make defconfig
-if [[ -n ${EDITCONFIG:-} ]]; then
-    make menuconfig
-    make savedefconfig
+BUILDROOT_VERSION=2019.02.3
+if [[ -d src ]]; then
+    cd src || exit 1
+    if [[ $(git describe --exact-match --tags HEAD) != "${BUILDROOT_VERSION}" ]]; then
+        git fetch origin ${BUILDROOT_VERSION}
+        git checkout ${BUILDROOT_VERSION}
+    fi
+else
+    git clone -b ${BUILDROOT_VERSION} git://git.busybox.net/buildroot src
+    cd src || exit 1
 fi
 
-# Build images
-make -j"$(nproc)"
+# Build the images for the architectures requested
+for CONFIG in "${CONFIGS[@]}"; do
+    # Clean up artifacts from the last build
+    make clean
 
-# Make sure images folder exists
-IMAGES_FOLDER=../../images/${1}
-[[ ! -d ${IMAGES_FOLDER} ]] && mkdir -p "${IMAGES_FOLDER}"
-
-# Copy new images
-# Make sure images exist before moving them
-IMAGES=( "output/images/rootfs.cpio" "output/images/rootfs.ext4" )
-for IMAGE in "${IMAGES[@]}"; do
-    if [[ ! -f ${IMAGE} ]]; then
-        echo "${IMAGE} could not be found! Did the build error?"
-        exit 1
+    BR2_DEFCONFIG=${CONFIG} make defconfig
+    if [[ -n ${EDITCONFIG:-} ]]; then
+        make menuconfig
+        make savedefconfig
     fi
-    cp -v "${IMAGE}" "${IMAGES_FOLDER}"
+
+    # Build images
+    make -j"$(nproc)"
+
+    # Make sure images folder exists
+    IMAGES_FOLDER=../../images/$(basename "${CONFIG//.config}")
```

Comment:
was the double `//` intentional?

Reply:
Yes, it's a habit (`/` removes the first occurrence, `//` removes all occurrences). Here it obviously does not matter so I can change it if it helps make it clearer.

Reply:
sorry, what do you mean by "removes" ?

Reply:
Sorry, removes the part after the slashes from the variable's value (an example makes it clearer):

```
$ CONFIG=arm.config.config; echo ${CONFIG/.config}
arm.config
$ CONFIG=arm.config.config; echo ${CONFIG//.config}
arm
```

Reply:
oh, maybe create a new variable on the line before use with a comment; I feel like I'll forget that's what `//` does in the future. Thanks for the explanation.